### PR TITLE
[FA] - Ch2 - P1 and P2

### PR DESCRIPTION
**Pull request by kambizG**
Two sections are translated and all comments for the first round of review are applied!

### CODE REVIEW

**Comment by kambizG** [comment]
Contents: 
Ch3 - P1 - reviewed by @hamedonline  
Ch3 - P2 - reviewed by @jowharshamshiri  
G - 11 new phrases

**Comment by jowharshamshiri** [comment]
@kambizG jan I did a quick check of a few lines and these are the issues that jump out to me:

- Your are using the Persian for "sorting" in place of "collation". I think "collate" is used to describe combining items in the original page.
 
- You're using the G equivalent of application features in place of the G equivalent of model features, which are not the same. 

- I can't find the word "sequential model" you added to G anywhere on page 2, I vaguely remember there was a mistranslated line in the previous review that had a phrase like it inside but I can't find it rn. 

- We still haven't had the chance to discuss the word "padding". I think it doesn't make sense to have a singe-word equivalent for it. What you describe with the very contrived equivalent("هم طول سازی", "make the same length") is the effect that padding has in the cases on this page, not the action that padding is in a general sense. Padding can be used to achieve other effects that your equivalent does not adequately describe. Instead of هم طول سازی we should go with full sentences like طول فلان را یکسان کنیم or  طول المان‌ها را به اندازه بزرگترین المان برسانیم and so on and so forth.
 
- A couple of places where the "می" prefix is not separated.

I think I've already contributed what I can. These were already [pointed out in our previous review](https://github.com/huggingface/course/pull/155) and this tells me there are probably more like them. I know you have fixed a lot of those issues, and some can be postponed to a later draft, which we can review later on. Just pointed these out bc the first two at least seem important. Peace and love brother.

**Review by jowharshamshiri — CHANGES_REQUESTED** [review]
See above.

**Comment by kambizG** [comment]
@lewtun ready to be merged!

**Comment by kambizG** [comment]
> See above.

Thanks man! 
Applied your comments. Let's discuss "هم‌طول‌سازی" and "collate" and apply changes later.

**Comment by jowharshamshiri** [comment]
> > See above.
> 
> Thanks man! Applied your comments. Let's discuss "هم‌طول‌سازی" and "collate" and apply changes later.

Kambiz Jan,

ON POINT 1:

DataCollatorWithPadding is not an OBJECT either. It is a CLASS. Use the Persian equivalent for class please.

ON POINT 3, I again went and checked the original text. When the text says:
```
we can inspect the `features` of our `raw_train_dataset`.
```
Please do not translate "features". You can not translate code. Translation guidelines. That is the name of an attribute for an object and references the code below that line.

**Comment by jowharshamshiri** [comment]
@kambizG 

ین قابلیتی را توسط کلاس `DataCollatorWithPadding` به ما می‌دهد. به محض این که این کلاس را تعریف کنیم


You changed from object to class in multiple places and again this doesn't make sense. You are not declaring this class. You are instantiating an object from this class. Please at least change to 

محض این که شیءای از این کلاس را تعریف کنیم

**Comment by lewtun** [comment]
Hi @kambizG and @jowharshamshiri - thanks to both of you for creating and reviewing the translations! Do I understand correctly that this shouldn't be merged yet? 

If not, feel free to ping me when it is :)

**Comment by kambizG** [comment]
> Hi @kambizG and @jowharshamshiri - thanks to both of you for creating and reviewing the translations! Do I understand correctly that this shouldn't be merged yet?
> 
> If not, feel free to ping me when it is :)


Hi @lewtun, 
we're done!

**Review by jowharshamshiri — APPROVED** [review]
This is a good place to end this review. Thanks @kambizG jan for bearing with me through the back and forth. ❤️

**Comment by lewtun** [comment]
Awesome job @kambizG and @jowharshamshiri - merging now!
